### DISCUSSION TITLE
Calibration troubleshooting

### DISCUSSION THREAD

**Discussion by electro991** [CrashOverride85/zc95#101] [Q&A] [answered]
Good afternoon,
I have finished assembling and trying first power up. Unfortunately calibration did not succed. OK is flashing and 9v goes off.

This is the debug output I a getting thru ttl on the output board. I would really appreciate if you had advice for troubleshooting. 

`ZC624 startup, firmware version: v1.9
CI2cSlave()
00 .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .
10 .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .
20 .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .
30 .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .
40 .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .
50 .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .
60 @  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .
70 .  .  .  .  .  .  .  .  .  .  .  .  .  .  .  .
Done.
core1_entry()
CMsgDac::CMsgDac()
COutput()
CPulseQueue::CPulseQueue(): ChannelIsolation: 1
COutputChannel(6,0)
COutputChannel(8,1)
COutputChannel(10,2)
COutputChannel(12,3)
calibrate for sm=0 FAILED! final voltage = 0.074927, dac_value = 2400 (expecting 0.075v - 0.090v)
calibrate for sm=1 FAILED! final voltage = 0.068481, dac_value = 2400 (expecting 0.075v - 0.090v)
calibrate for sm=2 FAILED! final voltage = 0.074121, dac_value = 2400 (expecting 0.075v - 0.090v)
calibrate for sm=3 FAILED! final voltage = 0.072510, dac_value = 2400 (expecting 0.075v - 0.090v)
One or more chanel failed calibration, not enabling power.
CMessageProcess()
HALT.
`

**Comment by CrashOverride85** [comment]
That looks like it might be the same issue I've heard some people are having with the mk312bt where it gives an "error 20" due to the current batch of IRF9Z24NPBF mosfets having a slightly different gate threshold voltage to what was expected. 
I suspect it's a similar reason here as that looks like its _very_ close to working.

I've extended the range it'll except in the [v1.9_zc624_extended_cal_range](https://github.com/CrashOverride85/zc95/tree/v1.9_zc624_extended_cal_range) branch, and built this firmware from it:
[OutputZc.uf2.zip](https://github.com/user-attachments/files/15943861/OutputZc.uf2.zip)
Can you try that on the output board, at let me know if it works, or changes anything?

**Reply by electro991** [reply, marked as answer]
Thank you so much ! It works and calibrates properly with this firmware !! Thank you for what you did. Very much appreciated and great project.

I am positive that was the issue as prior to updating, I replaced both 0.5R resistors and re-worked all solder on the board just in case. 

For the record, the IRF9Z24NPBF I have the following imprint on them; **P226M** date code, assembly lot code **AQ**
